tests/common.bash

```diff
@@ -623,12 +623,13 @@ function install_cni_plugins() {
 
 # base_version: The version to be intalled in the ${major}.${minor} format
 function install_cri_containerd() {
+	set -x
 	base_version="${1}"
 
 	project="containerd/containerd"
 	version=$(get_latest_patch_release_from_a_github_project "${project}" "${base_version}")
 
-	tarball_name="cri-containerd-cni-${version//v}-linux-$(${repo_root_dir}/tests/kata-arch.sh -g).tar.gz"
+	tarball_name="containerd-${version//v}-linux-$(${repo_root_dir}/tests/kata-arch.sh -g).tar.gz"
```

Comment:
I think this line has broken the runners.
If I compare before:
```
# sudo tar -xvf cri-containerd-cni-1.6.36-linux-amd64.tar.gz
etc/
etc/systemd/
etc/systemd/system/
etc/systemd/system/containerd.service
etc/crictl.yaml
etc/cni/
etc/cni/net.d/
etc/cni/net.d/10-containerd-net.conflist
usr/
usr/local/
usr/local/sbin/
usr/local/sbin/runc
usr/local/bin/
usr/local/bin/containerd-shim
usr/local/bin/ctr
usr/local/bin/containerd-shim-runc-v1
usr/local/bin/ctd-decoder
usr/local/bin/containerd-stress
usr/local/bin/containerd
usr/local/bin/critest
usr/local/bin/crictl
usr/local/bin/containerd-shim-runc-v2
opt/
opt/cni/
opt/cni/bin/
opt/cni/bin/bandwidth
opt/cni/bin/host-local
opt/cni/bin/portmap
opt/cni/bin/tuning
opt/cni/bin/loopback
opt/cni/bin/macvlan
opt/cni/bin/bridge
opt/cni/bin/host-device
opt/cni/bin/vlan
opt/cni/bin/vrf
opt/cni/bin/firewall
opt/cni/bin/ptp
opt/cni/bin/dhcp
opt/cni/bin/ipvlan
opt/cni/bin/static
opt/cni/bin/sbr
opt/containerd/
opt/containerd/cluster/
opt/containerd/cluster/version
opt/containerd/cluster/gce/
opt/containerd/cluster/gce/env
opt/containerd/cluster/gce/cni.template
opt/containerd/cluster/gce/configure.sh
opt/containerd/cluster/gce/cloud-init/
opt/containerd/cluster/gce/cloud-init/master.yaml
opt/containerd/cluster/gce/cloud-init/node.yaml
```
with after:
```
# sudo tar -xvf containerd-1.6.36-linux-amd64.tar.gz -C temp
bin/
bin/containerd-shim
bin/ctr
bin/containerd-shim-runc-v1
bin/containerd-stress
bin/containerd
bin/containerd-shim-runc-v2
```
then you can see the file paths are quite different. I think that as we extract to `/` we are overriding the `/bin` directory and bricking any system we install this on:
```
# sudo tar -xvf containerd-1.6.36-linux-amd64.tar.gz -C /
bin/
bin/containerd-shim
bin/ctr
bin/containerd-shim-runc-v1
bin/containerd-stress
bin/containerd
bin/containerd-shim-runc-v2
root@sh-roks-test1:~# ls /bin
containerd  containerd-shim  containerd-shim-runc-v1  containerd-shim-runc-v2  containerd-stress  ctr
```

Reply:
Hi  @stevenhorsman 

I finally got what's wrong with change now.  Since our os system would make a soft link from bin to usr/bin, as below:

![1](https://github.com/user-attachments/assets/a6c854c0-87a1-4943-b082-6f82e2f3dc69)

Then the "tar -xvf" in our CI code: https://github.com/kata-containers/kata-containers/blob/main/tests/common.bash#L634  would override the softlink file. Since the containerd tarball contains the following files:

bin/
bin/containerd-shim
bin/ctr
bin/containerd-shim-runc-v1
bin/containerd-stress
bin/containerd
bin/containerd-shim-runc-v2

thus it would override the softlink "bin" and created a real directory file "bin", which would make the binaries missing from the /bin directory. 

So in order to prevent the override happen, we should add "-h, --dereference          follow symlinks; archive and dump the files they" to the tar command.

Reply:
Great, thanks for the investigation. I guess post 3.12 release we could fix this in #10665 and then I'm happy to test it out manually before we open that PR and run it on the self-hosted runners?

Reply:
> So in order to prevent the override happen, we should add "-h, --dereference follow symlinks; archive and dump the files they" to the tar command.

@lifupan `--dereference` affects the creation of the tarball IIUC. For the unpacking you rather want to use :
```
       --keep-directory-symlink
              Don't replace existing symlinks to directories when extracting.

```

Reply:
@stevenhorsman  @gkurz  Ok, I'll test it locally first, and then I'll reopen the PR https://github.com/kata-containers/kata-containers/pull/10665 with the "--keep-directory-symlink" for tar cmd once the 3.12 released.